snapshot-labs/sx-monorepo
# fix: avoid voting power computation for starknet account on snapshot spaces

When visiting snapshot spaces, when connected with a starknet account, the voting power fetching will fail, due to score-api returning an error (invalid address), as starknet address are not supported on score-api.

This PR will skip the voting power fetching from score-api when addresses are not supported, and will show 0 VP for all strategies instead.

Closes: #155 

### How to test

1. Go to http://localhost:8080/#/s:stgdao.eth/proposals
2. Open NETWORK tab in dev console
3. Connect with a starknet account
4. It will show that you have 0 voting power
5. In the network TAB, there are no requests to https://score.snapshot.org/
6. Open the voting power modal, it should list all strategies, with 0 VP, like 

![Screenshot 2024-03-08 at 16 28 06](https://github.com/snapshot-labs/sx-monorepo/assets/495709/636158b6-a353-4f5c-a0fb-2355bf6c57b1)


### Other changes

This PR also refactor the `getVotingPower` function a little bit, to list all strategies with 0 VP, in case `getOffchainStrategy(name)` returns nothing, instead of just the first strategy.
